File: leetcode-python-go/1704. 判断字符串的两半是否相似/main.go
package main

import (
	"fmt"
	"strings"
)

func halvesAreAlike(s string) bool {
	cnt := 0
	for _, c := range s[:len(s)/2] {
		if strings.ContainsRune("aeiouAEIOU", c) {
			cnt++
		}
	}
	for _, c := range s[len(s)/2:] {
		if strings.ContainsRune("aeiouAEIOU", c) {
			cnt--
		}
	}
	return cnt == 0
}

func main() {
	s := "textbook"
	results := halvesAreAlike(s)
	fmt.Println(results)
}

File: leetcode-python-go/1704. 判断字符串的两半是否相似/main.py
s = "textbook"

def halvesAreAlike(s: str) -> bool:
	VOWELS = "aeiouAEIOU"
	a, b = s[:len(s) // 2], s[len(s) // 2:]
	return sum(c in VOWELS for c in a) == sum(c in VOWELS for c in b)

print(halvesAreAlike(s))